identity-wallet/src/state/credentials/mod.rs

```diff
@@ -54,45 +59,90 @@ impl TryFrom<serde_json::Value> for VerifiableCredentialRecord {
-                            "credentialSubject": credential_subject_value,
-                        }
+            // Try to parse the Verifiable Credential as an SD-JWT credential.
+            let (id, format, data, issuance_date) = if let Some(sd_jwt_vc) = verifiable_credential
+                .as_str()
+                .and_then(|verifiable_credential| verifiable_credential.parse::<SdJwtVc>().ok())
+            {
+                info!("Verifiable Credential parsed as a SD-JWT VC");
+
+                let issuance_date = sd_jwt_vc.claims().iat.map(|iat| iat.to_rfc3339()).unwrap_or_default();
+                let mut credential_subject = sd_jwt_vc
+                    .clone()
+                    .into_disclosed_object(&Sha256Hasher::new())
+                    .map_err(|_| AppError::Error("Failed to convert SD JWT VC to Disclosed Object".to_string()))?;
+
+                // Remove the SD-JWT specific fields that should not be displayed in the frontend.
+                for key in ["iss", "nbf", "exp", "status", "iat", "sub", "_sd_alg", "cnf", "vct"] {
+                    credential_subject.remove(key);
+                }
+
+                // TODO: We are using this hash as Credential ID so that we can prevent credential duplication in
+                // demo situations. Now we can actually delete Credentials in UniMe we don't need to use the hash of the
+                // credential as the ID anymore. We should simply generate a random UUID.
+                let hash = sha256::digest(json!(credential_subject).to_string());
+
+                let id = Uuid::from_slice(&hash.as_bytes()[..16])?.to_string();
+
+                let format = CredentialFormats::VcSdJwt(());
+                // TODO: Remove this workaround that is basically a way of disguising the SD JWT VC as a VC so that
+                // it can be displayed in the Frontend.
+                let data = json!({
+                    "type": ["VerifiableCredential"],
+                    "issuer": sd_jwt_vc.claims().iss,
+                    "credentialSubject": credential_subject
+
+                });
+
+                (id, format, data, issuance_date)
+            } else {
+                let credential_display = get_unverified_jwt_claims(&verifiable_credential)?["vc"].clone();
+
+                // TODO: We are using this hash as Credential ID so that we can prevent credential duplication in
+                // demo situations. Now we can actually delete Credentials in UniMe we don't need to use the hash of the
+                // credential as the ID anymore. We should simply generate a random UUID.
+                // Derive the hash from the credential display.
+                let hash = {
+                    let type_value = credential_display["type"].clone();
+
+                    let mut credential_subject_value = credential_display["credentialSubject"].clone();
+
+                    // TODO(ngdil): Remove this hard-coded logic.
```

Comment:
Can this be removed now?

Reply:
Not in this current form, but we could simply start using random UUIDs instead of generating them from the Hash wyt? (see comment above)

Reply:
I meant if we can get rid of popping the `"Passport Number"`, `"Staff Number"` etc. out of the credential, since this code got touched in this PR as well. But I guess we make it a seperate issue to remove all hard-coded demo logic.